ci/run.ps1

```diff
@@ -54,6 +54,11 @@ echo "Extracting ci_data.zip"
 Expand-Archive $env:AGENT_WORKFOLDER\ci_data.zip -DestinationPath $env:AGENT_WORKFOLDER\ci_data
 & "$env:AGENT_WORKFOLDER\ci_data\install.ps1" -Arch $env:arch -Compiler $env:compiler -Boost $true -DMD $dmd
 
+DownloadFile -Source https://downloads.python.org/pypy/pypy3.8-v7.3.9-win64.zip -Destination $env:AGENT_WORKFOLDER\pypy38.zip
+Expand-Archive $env:AGENT_WORKFOLDER\pypy38.zip -DestinationPath $env:AGENT_WORKFOLDER\pypy38
+$ENV:Path = $ENV:Path + ";$ENV:AGENT_WORKFOLDER\pypy38\pypy3.8-v7.3.9;$ENV:AGENT_WORKFOLDER\pypy38\pypy3.8-v7.3.9\Scripts"
+pypy3 -m ensurepip
```

Comment:
It seems like azure has a hostedtoolscache version of PyPy already:
https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#pypy
https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#pypy

Should we use this instead?

Reply:
That should work. We need to add it to the path. Do you know where it is installed?

Reply:
Based on:

https://github.com/mesonbuild/meson/blob/e5aa47d8af917282794a8da3956ce444b8d8378d/azure-pipelines.yml#L62-L67

That key for "addToPath" seems like this is handled automatically for us, if you use the Microsoft task for adding a tool.

I assume either there is a similar task type for PyPy, or some version string instead of "3.7" that denotes to use PyPy.

Unfortunately I know very little about the azure pipelines ecosystem.

Reply:
right, but that will get us only pypy3 or only python3.7.

Reply:
maybe we can add both steps?

One caveat: in the PyPy directory there are a few *.exe files, including python.exe, so we need to make sure the PyPy directory is last on the PATH so it does not override CPython
```
>dir *.exe /b
pypy.exe
pypy3.9.exe
pypy3.9w.exe
pypyw.exe
python.exe
pythonw.exe
```

Reply:
Is it not possible to use this twice and get two versions of a tool into the path? Do they just assume no one ever does that other than once per action in a matrix?

Reply:
Looking at the [source of the task](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/UsePythonVersionV0/usepythonversion.ts#L125), the name seems to suggest it will prepend. So I will try adding the PyPy step before the CPython one, in the hopes that does the RightThing